extension/app/ts/components/simulationExplaining/customExplainers/SimpleTokenApprovalVisualisation.tsx

```diff
@@ -0,0 +1,63 @@
+import { RenameAddressCallBack } from '../../../utils/user-interface-types.js'
+import { SimulatedAndVisualizedTransaction } from '../../../utils/visualizer-types.js'
+import { BigAddress } from '../../subcomponents/address.js'
+import { Token721AmountField, TokenAmount, TokenSymbol } from '../../subcomponents/coins.js'
+import { GasFee } from '../SimulationSummary.js'
+
+export function SimpleTokenApprovalVisualisation({ transaction, renameAddressCallBack }: { transaction: SimulatedAndVisualizedTransaction, renameAddressCallBack: RenameAddressCallBack }) {
+	if (transaction.to === undefined) throw new Error('Contract creation trasaction')
+	if (transaction.tokenResults.length != 1) throw new Error('Multiple token events transfer')
+
+	const approval = transaction.tokenResults[0]
+	if (!approval.isApproval) throw new Error('Is Not Approval')
```

Comment:
Where is this error caught?  Is it presented?

Reply:
this should not never occur as its tested beforehand, so its not caught anywhere. If its thrown, something has wrong elsewhere. I guess one option would be to add it to the typing so we wouldn't need to check it

Reply:
There are several errors being thrown, are these all critical programmer errors and not something we should reasonably be able to deal with?

Reply:
which can be fixed with typings, but I was lazy :D

Reply:
Recommending fixing with type checking then.  😀